File: maxWaterContainer.js
class Solution {
  /**
   * @param {number[]} heights
   * @return {number}
   */
  maxArea(heights) {
    let res = 0;
    let l = 0,
      r = heights.length - 1;

    while (l < r) {
      let water = Math.min(heights[l], heights[r]) * (r - l);
      res = Math.max(water, res);

      if (heights[l] <= heights[r]) {
        l++;
      } else {
        r--;
      }
    }
    return res;
  }
}

File: max_water_container.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l, r = 0, len(heights) - 1
        res = 0
        while l < r:
            length = r - l
            water = min(heights[l], heights[r]) * length
            res = max(res, water)
            if heights[l] <= heights[r]:
                l += 1
            else:
                r -= 1
        return res
